src/platforms/bool.rkt

```diff
@@ -29,11 +29,19 @@
 (define (or-fn . as)
   (ormap identity as))
 
-(define-operator-impl (not not bool) bool [fl not])
+(define-operator-impl (not [x : bool]) bool #:spec (not x) #:fpcore (! (not x)) #:fl not)
 
-(define-operator-impl (and and bool bool) bool [fl and-fn])
+(define-operator-impl (and [x : bool] [y : bool])
+                      bool
+                      #:spec (and x y)
+                      #:fpcore (! (and x x))
```

Comment:
Usually when the set of rounding properties is empty, `!` is not used. Should you update your syntax checker to handle that case?

Reply:
So should the fpcore be #:fpcore (and x x)? If so the syntax checker will handle it